crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -22,23 +22,104 @@ ghost_actor::ghost_chan! {
 
 pub(crate) struct KitsuneP2pActor {
     channel_factory: ghost_actor::actor_builder::GhostActorChannelFactory<Self>,
-    #[allow(dead_code)]
     internal_sender: ghost_actor::GhostSender<Internal>,
-    #[allow(dead_code)]
     evt_sender: futures::channel::mpsc::Sender<KitsuneP2pEvent>,
+    transport: ghost_actor::GhostSender<TransportListener>,
     spaces: HashMap<Arc<KitsuneSpace>, AsyncLazy<ghost_actor::GhostSender<KitsuneP2p>>>,
 }
 
+#[allow(clippy::type_complexity)]
```

Comment:
instead of allowing for high complexity, is it much harder to just break this out for better readability?

it really does make it so much harder to follow the code to its conclusions

Reply:
This is all rust nonsense... it should just be an async function - but they haven't figured out how to make async functions recursive yet... so you have to manually do all the pinning, boxing, and Future-ing... Maybe I can turn it into a macro or something...

Reply:
ah, I see, the recursion is the problematic bit... well, still just thinking it could become it's own function at each level of the match statements, and I think it could still utilize double-function style recursion?

Reply:
alas, recursion at any depth is invalid, the double function doesn't solve it.

But I was able to switch it to returning a MustBoxFuture<> type which simplified it enough that clippy stopped complaining.